### PR TITLE
library: Ported SpinButton in RUST

### PR DESCRIPTION
**Pull request by onkarrai06**
Tried to port SpinButton, works fine on my setup, please check.

### CODE REVIEW

File: src/Library/demos/Spin Button/code.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+use glib::clone;
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::SpinType::StepForward;
+
+pub fn main() {
+    let hours: gtk::SpinButton = workbench::builder().object("hours").unwrap();
+    let minutes: gtk::SpinButton = workbench::builder().object("minutes").unwrap();
+
+    fn tellTime(hours: &str, minutes: &str) {
```
Comment (Hofer-Julian) [inline src/Library/demos/Spin Button/code.rs:11]: - Please keep this function in the global namespace, below of `main`.
- Also please rename it to `tell_time`

File: src/Library/demos/Spin Button/code.rs [hunk 2]
```diff
@@ -0,0 +1,41 @@
+use crate::workbench;
+use glib::clone;
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::SpinType::StepForward;
+
+pub fn main() {
+    let hours: gtk::SpinButton = workbench::builder().object("hours").unwrap();
+    let minutes: gtk::SpinButton = workbench::builder().object("minutes").unwrap();
```
Comment (Hofer-Julian) [inline src/Library/demos/Spin Button/code.rs:9]: Set those to "00" as with JavaScript.

File: src/Library/demos/Spin Button/code.rs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    });
+
+    minutes.connect_output(move |minutes| {
+        let value = minutes.value();
+        let text = format!("{:02}", value);
+        minutes.set_text(&text);
+        return true.into();
+    });
+
+    minutes.connect_wrapped(clone!(@weak hours => move |minutes| {
```
Comment (Hofer-Julian) [inline src/Library/demos/Spin Button/code.rs:33]:
```suggestion
    minutes.connect_wrapped(clone!(@weak hours => move |_| {
```

File: src/Library/demos/Spin Button/code.rs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      let spin_type: gtk:: SpinType = StepForward;
+      hours.spin(spin_type, 1.0);
```
Comment (Hofer-Julian) [inline src/Library/demos/Spin Button/code.rs:35]:
```suggestion
      hours.spin(SpinType::StepForward, 1.0);
```

File: src/Library/demos/Spin Button/code.rs [hunk 5]
```diff
@@ -0,0 +1,41 @@
+use crate::workbench;
+use glib::clone;
+use gtk::glib;
+use gtk::prelude::*;
+use gtk::SpinType::StepForward;
```
Comment (Hofer-Julian) [inline src/Library/demos/Spin Button/code.rs:5]:
```suggestion
use gtk::SpinType;
```

**Review by Hofer-Julian — COMMENTED** [review]
Apart from the comments, it looks good

**Comment by onkarrai06** [comment]
Fixed, please check again.
@Hofer-Julian

**Review by Hofer-Julian — APPROVED** [review]
Made small improvements in https://github.com/sonnyp/Workbench/pull/568/commits/9c404d00b67ac5d7cd3986c76404d49a0b6d7090

Thanks!

**Pull request merged**